engine/table/src/main/java/io/deephaven/engine/table/impl/select/DynamicWhereFilter.java

```diff
@@ -64,13 +66,22 @@ public class DynamicWhereFilter extends WhereFilterLivenessArtifactImpl
     private RecomputeListener listener;
     private QueryTable resultTable;
 
+    /**
+     * Construct a DynamicWhereFilter with key values from given set table. The set table may be static or refreshing.
+     * <p>
+     * NOTE: If the set table is refreshing, it must not contain duplicate keys for the given
```

Comment:
This seems like an argument for moving the `setTable` preparation from `QueryTable.whereInInternal` down to this constructor. Then we wouldn't need this documentation, and the landmine is no longer present.